### PR TITLE
Feat/gas

### PR DESCRIPTION
**Pull request by nikkolasg**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -17,7 +18,8 @@ function App () {
   console.log('reloaded')
   const [miners, setMiners] = useState()
   const [head, setHead] = useState()
-  const [node, setNode] = useState('wss://node.glif.io/space07/lotus/rpc/v0')
+  //const [node, setNode] = useState('wss://node.glif.io/space07/lotus/rpc/v0')
+  const [node, setNode] = useState('wss://lotus.jimpick.com/spacerace_api/0/node/rpc/v0')
```
Comment (nicola) [inline src/App.js]: we should switch this back, you can always switch it with the select menu

File: src/App.js [hunk 2]
```diff
@@ -104,23 +106,28 @@ function App () {
...
-            <Full client={client} miners={miners} />
-          </Route>
-          <Route path='/status'>
-            <Status head={head} spa={spa} client={client} miners={miners} />
-          </Route>
-          <Route path='/'>
-            <Home miners={miners} />
-          </Route>
-        </Switch>
+        <div className="container">
```
Comment (nicola) [inline src/App.js:121]: we should remove this, each View should have it's own container by default I think (?) if not, then ok

File: src/services/filecoin/index.js [hunk 3]
```diff
@@ -165,6 +165,8 @@ export default class Filecoin {
     const provider = new BrowserProvider(endpointUrl)
     console.log('new endpoint', endpointUrl)
     this.client = new LotusRPC(provider, { schema })
+    this.parents = {}
+    this.receipts = {}
```
Comment (nicola) [inline src/services/filecoin/index.js:200]: genius, we should do caching for all other calls!

File: src/views/Home.js [hunk 4]
```diff
@@ -22,6 +22,9 @@ export default function Home ({ miners }) {
         See deadlines of <Link to='/full'>top 50 miners</Link> or click on
         individual miners or the <Link to='/status'>network status</Link>.
       </div>
+      <div className="row">
+        See <Link to='/gas'> here </Link> for a detailled gas analysis.
```
Comment (nicola) [inline src/views/Home.js]: detailed

File: src/App.js (hunk 2 above)
Comment (nikkolasg) [reply src/App.js:121]: The whole switch statement was *not* inside a container, only the top bar, the logo that's why i added one.

**Comment by nikkolasg** [comment]
Should be ready now @nicola

File: src/App.js [hunk 5]
```diff
@@ -64,7 +66,7 @@ function App () {
       if (mounted) {
         fetchingHead()
       }
-    }, 5000)
+    }, 15000)
```
Comment (nicola) [inline src/App.js]: why?

File: src/views/Gas.js [hunk 6]
```diff
@@ -0,0 +1,395 @@
...
+    const setCollapseEvent = pie => {
+        // XXX This is bad because it requires names of ID to be constant but 
+        // it'll do for now
+        $("#collapseBiggest").on('show.bs.collapse', function () {
+            pie.update();
+        });
+        $("#collapseBiggest").on('hide.bs.collapse', function () {
+            pie.update();
+        });
+    }
```
Comment (nicola) [inline src/views/Gas.js]: is this jquery? not sure this is the right design pattern, we should fix this (things may not work as expected)

File: src/views/Gas.js [hunk 7]
```diff
@@ -0,0 +1,395 @@
...
+        }
+        const newDataset = { 
+            datasets: datasets,
+            labels: labels,
+        }
+        setOpts(newOpts)
+        setData(newDataset)
+
+        if (pie == undefined) {
+            var ctx = document.getElementById(id).getContext('2d');
```
Comment (nicola) [inline src/views/Gas.js]: same here, not sure 100% this is the right way

File: src/App.js (hunk 5 above)
Comment (nikkolasg) [reply src/App.js]: it's causing to refresh more than necessary it seems  and polluting the logs a bit but it's fine - added it back

**Comment by nikkolasg** [comment]
@nicola Indeed there's a way to use ref ! Not sure it would have caused problems anyway since jquery didnt touch the dom structure itself but looks much better like this, thanks for the suggestion!

**Comment by nicola** [comment]
Hey when I was on gas I had a full page error and the miner search seems not to work.

Happens to you too?

**Pull request merged**